Source/AwsCommonRuntimeKit/io/HostResolver.swift

```diff
@@ -4,98 +4,93 @@
 import AwsCCommon
 import AwsCIo
 
-public typealias HostResolvedContinuation = CheckedContinuation<[HostAddress], Error>
-
-public protocol HostResolver: AnyObject {
-    var rawValue: UnsafeMutablePointer<aws_host_resolver> { get set }
-    var config: UnsafeMutablePointer<aws_host_resolution_config> { get }
-    func resolve(host: String) async throws -> [HostAddress]
-}
-
-public final class DefaultHostResolver: HostResolver {
-    public var rawValue: UnsafeMutablePointer<aws_host_resolver>
-    public var config: UnsafeMutablePointer<aws_host_resolution_config>
+public class HostResolver {
+    let rawValue: UnsafeMutablePointer<aws_host_resolver>
+    let maxTTL: Int
     private let allocator: Allocator
 
-    public init(eventLoopGroup elg: EventLoopGroup,
-                maxHosts: Int,
-                maxTTL: Int,
-                allocator: Allocator = defaultAllocator,
-                shutdownCallback: ShutdownCallback? = nil) throws {
-        let shutdownCallbackCore = ShutdownCallbackCore(shutdownCallback)
+    public static func makeDefault(eventLoopGroup: EventLoopGroup,
+                                   maxHosts: Int,
+                                   maxTTL: Int,
+                                   allocator: Allocator = defaultAllocator,
+                                   shutdownCallback: ShutdownCallback? = nil) throws -> HostResolver {
+        try HostResolver(eventLoopGroup: eventLoopGroup, maxHosts: maxHosts, maxTTL: maxTTL, allocator: allocator, shutdownCallback: shutdownCallback)
+    }
+
+    init(eventLoopGroup: EventLoopGroup,
+         maxHosts: Int,
+         maxTTL: Int,
+         allocator: Allocator = defaultAllocator,
+         shutdownCallback: ShutdownCallback? = nil) throws {
         self.allocator = allocator
+        let shutdownCallbackCore = ShutdownCallbackCore(shutdownCallback)
         guard let rawValue: UnsafeMutablePointer<aws_host_resolver> = withUnsafePointer(
                 to: shutdownCallbackCore.getRetainedShutdownOptions(), { shutdownCallbackCorePointer in
-            var options = aws_host_resolver_default_options(max_entries: maxHosts,
-                    el_group: elg.rawValue,
-                    shutdown_options: shutdownCallbackCorePointer,
-                    system_clock_override_fn: nil)
-            //TODO: use const in C-IO to avoid mutable pointer here
-            return withUnsafeMutablePointer(to: &options) { aws_host_resolver_new_default(allocator.rawValue, $0) }
+            var options = aws_host_resolver_default_options()
+            options.max_entries = maxHosts
+            options.el_group = eventLoopGroup.rawValue
+            options.shutdown_options = shutdownCallbackCorePointer
+            return aws_host_resolver_new_default(allocator.rawValue, &options)
```

Comment:
Similar to a comment above, can we define this logic in a `getRawValue` closure to make the `guard` statement easier to read.

Reply:
Thanks, fixed.